Amplify/Core/Support/AsyncChannel.swift

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+actor AsyncChannel<Element: Sendable>: AsyncSequence {
```

Comment:
nit: Call it `AmplifyAsyncChannel` ?

Reply:
there is reason to specialize this name

Reply:
I think if developer uses both Amplify and AsyncAlgorithms this names can conflict. We cannot use module name to distinguish between them because of the known issue in SPM where it cannot figure out the type `Amplify` vs the module `Amplify`. https://github.com/apple/swift/issues/43510